### PR TITLE
Issue #12168: failure of update github.io to clone checkstyle.github.io

### LINKED ISSUE #12168: failure of update github.io to clone checkstyle.github.io
https://github.com/checkstyle/checkstyle/runs/8179173614?check_suite_focus=true
failure of update github.io
```
Clone by ssh only to avoid passwords on push ...
Cloning into 'checkstyle.github.io'...
Warning: Permanently added the ECDSA host key for IP address '140.82.114.3' to the list of known hosts.
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
Cloning into 'checkstyle.github.io'...
Warning: Permanently added the ECDSA host key for IP address '140.82.113.3' to the list of known hosts.
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
Cloning into 'checkstyle.github.io'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
Cloning into 'checkstyle.github.io'...
Warning: Permanently added the ECDSA host key for IP address '140.82.114.4' to the list of known hosts.
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
Cloning into 'checkstyle.github.io'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
./.ci/update-github-io.sh: line 23: cd: .ci-temp/checkstyle.github.io: No such file or directory
Error: Process completed with exit code 1.
```

### PR DESCRIPTION
**Pull request by Rahulkhinchi03**
Fixes #12168

### CODE REVIEW

File: .github/workflows/update_github_io.yml [hunk 1]
```diff
@@ -15,18 +15,33 @@ jobs:
...
         run: |
           ./.ci/update-github-io.sh ${{ github.event.inputs.version }}
-      - name: Commit and Push
+      - name: Commit
         run: |
-          git config --global user.name 'github-actions[bot]'
-          git config --global user.email 'github-actions[bot]@users.noreply.github.com'
-          cd . ci-temp/checkstyle.github.io
+          cd checkstyle.github.io
+          git rm -rf *
```
Comment (rnveach) [inline .github/workflows/update_github_io.yml:38]: We are really removing all files?

Comment (Rahulkhinchi03) [reply .github/workflows/update_github_io.yml:38]: I have taken the reference from `release.sh` -> https://github.com/checkstyle/checkstyle/blob/master/release.sh#L131

**Comment by Rahulkhinchi03** [comment]
@romani please review!

File: .github/workflows/update_github_io.yml (hunk 1 above)
Comment (romani) [reply .github/workflows/update_github_io.yml:38]: Yes, remove all, to have no history in repo, to make repo quick to checkout

File: .github/workflows/update_github_io.yml [hunk 2]
```diff
@@ -18,15 +18,30 @@ jobs:
         uses: actions/checkout@v3
         with:
           fetch-depth: 0
+          token: ${{ secrets.PAT }}
+      - name: Make Temp Folder
+        run: |
+          mkdir .ci-temp
+          cd .ci-temp
```
Comment (romani) [inline .github/workflows/update_github_io.yml:25]: cd is not persistent between "run", we already discussed this.

File: .github/workflows/update_github_io.yml [hunk 3]
```diff
@@ -18,15 +18,30 @@ jobs:
...
+      - name: Checkout io repo
+        uses: actions/checkout@v3
+        with:
+          token: ${{ secrets.PAT }}
+          repository: checkstyle/checkstyle.github.io
+          path: checkstyle.github.io
```
Comment (romani) [inline .github/workflows/update_github_io.yml:31]: checkout directly to .ci-temp/checkstyle.github.io

File: .github/workflows/update_github_io.yml (hunk 1 above)
Comment (romani) [reply .github/workflows/update_github_io.yml:38]: I do not see copy of new files to repo
https://github.com/checkstyle/checkstyle/blob/7658776d784d82531db387027d5bfb1cb7cde0a9/release.sh#L133

File: .github/workflows/update_github_io.yml [hunk 4]
```diff
@@ -18,15 +18,30 @@ jobs:
...
+          git checkout HEAD -- CNAME
           git config user.name 'github-actions[bot]'
           git config user.email 'github-actions[bot]@users.noreply.github.com'
           git add . && git commit -m "release ${{ github.event.inputs.version }}"
+      - name: Push Changes
+        run: |
           echo "Push site content to remote ..." 
           echo "We do force to avoid history changes as github.io shows only HEAD." 
-          git push origin --force 
+          git push origin HEAD:master --force
```
Comment (romani) [inline .github/workflows/update_github_io.yml:47]: Why `HEAD:master` is required ?
remove

**Review by romani — CHANGES_REQUESTED** [review]
items:

File: .github/workflows/update_github_io.yml (hunk 1 above)
Comment (Rahulkhinchi03) [reply .github/workflows/update_github_io.yml:38]: https://github.com/Rahulkhinchi03/checkstyle/runs/8253537139?check_suite_focus=true#step:5:14 didn't match any file

**Pull request closed without merging**